src/bundle.h

```diff
@@ -47,6 +55,13 @@ void track_bundle_in_statedir(const char *bundle_name, const char *state_dir);
  */
 void track_bundle(const char *bundle_name);
 
+/**
+ * @brief Validates that a tracking directory exists and that it is not
+ * empty, if it is, it creates it and marks all installed bundles as tracked.
+ * @param state_dir The path to the state directory.
+ */
+bool validate_tracking_dir(const char *state_dir);
```

Comment:
Exporting a validate_tracking_dir here is adds a great possibility of errors in the future. And it doesn't help with the problem you have on os-install that you needed the .init hack.

So when you are tracking a new bundle you are calling the validate_tracking_dir() implicitly and you need to prevent that by creating a file if you don't want to track.  And in the case of remove track files you need to explicitly call the initialize function. This is prone to errors because it's confusing for the users of the library if they need to call or not the initialize function.

There are for sure multiple solutions to this problem. What I would suggest is to call validate_tracking_dir() implicitly on create_state_dir() so we always know that the directory is valid when we are using any tracked function.

 - For bundle_remove and bundle_add we don't need to worry about this anymore 
 - We aren't going to be checking if track directory is valid every time we track a new bundle on os-install and bundle-add
 - On os-install we remove the track of bundles we don't need instead of tracking new bundles. The downside is that we are going to be creating several useless files, but as they are empty and we are doing that on install time, it's totally ok.
 - If we add any other function that uses tracked bundles we don't need to worry about this anymore